packages/core/src/components/fields/ObjectField.tsx

```diff
@@ -195,36 +195,39 @@ class ObjectField<T = any, S extends StrictRJSFSchema = RJSFSchema, F extends Fo
    * @param schema - The schema element to which the new property is being added
    */
   handleAddClick = (schema: S) => () => {
-    if (!schema.additionalProperties) {
+    if (!(schema.patternProperties || schema.additionalProperties)) {
```

Comment:
This matches the order in code down in utils
```suggestion
    if (!(schema.additionalProperties || schema.patternProperties)) {
```